osu.Game.Rulesets.Taiko/Skinning/Default/CirclePiece.cs

```diff
@@ -157,6 +160,13 @@ protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint,
             if (!effectPoint.KiaiMode)
                 return;
 
+            FlashBox
+                // Make sure the hit indicator usage of FlashBox doesn't get faded out prematurely by a kiai flash
+                .DelayUntilTransformsFinished()
```

Comment:
Not too keen on this, seems like a bit of a hacky approach.

How about

```diff
diff --git a/osu.Game.Rulesets.Taiko/Skinning/Default/CirclePiece.cs b/osu.Game.Rulesets.Taiko/Skinning/Default/CirclePiece.cs
index 70eacbe61d..277f674b03 100644
--- a/osu.Game.Rulesets.Taiko/Skinning/Default/CirclePiece.cs
+++ b/osu.Game.Rulesets.Taiko/Skinning/Default/CirclePiece.cs
@@ -4,6 +4,7 @@
 #nullable disable
 
 using System;
+using osu.Framework.Allocation;
 using osu.Framework.Audio.Track;
 using osu.Framework.Extensions.Color4Extensions;
 using osu.Framework.Graphics;
@@ -14,6 +15,7 @@
 using osu.Game.Graphics;
 using osu.Game.Graphics.Backgrounds;
 using osu.Game.Graphics.Containers;
+using osu.Game.Rulesets.Objects.Drawables;
 using osu.Game.Rulesets.Taiko.Objects;
 using osuTK.Graphics;
 
@@ -155,14 +157,18 @@ private void resetEdgeEffects()
             };
         }
 
+        [Resolved]
+        private DrawableHitObject drawableHitObject { get; set; }
+
         protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, ChannelAmplitudes amplitudes)
         {
             if (!effectPoint.KiaiMode)
                 return;
 
+            if (drawableHitObject.State.Value != ArmedState.Idle)
+                return;
+
             FlashBox
-                // Make sure the hit indicator usage of FlashBox doesn't get faded out prematurely by a kiai flash
-                .DelayUntilTransformsFinished()
                 .FadeTo(flash_opacity)
                 .Then()
                 .FadeOut(Math.Max(80, timingPoint.BeatLength - 80), Easing.OutSine);

```

Reply:
Seems good! I've made one small adjustment by removing the return if state isn't `Idle`, since it would cause the background edge effects to be unreachable in those situations (they do not seem to work at the moment anyway, I'll take a look into this later). Did a quick test in-game and works fine.